fix(FilesPicker): Prevent selection of nodes without create permission

artonge opened the pull request:
It is misleading to show those nodes in the file picker, as picking them will always result in an error.

Todo:

- [x] Update `@nextcloud/dialog` to uses the new API from https://github.com/nextcloud-libraries/nextcloud-dialogs/pull/2033

The new version will also prevent navigation to non-readable folders. This will prevent navigating to E2EE folders.

| Before | After |
|--------|--------|
| <img width="852" height="880" alt="image" src="https://github.com/user-attachments/assets/d06a25a6-e599-4d28-a62e-54692bd6e552" /> |<img width="852" height="880" alt="Screenshot From 2025-10-17 11-17-19" src="https://github.com/user-attachments/assets/f69f4966-bef2-47eb-a70e-6b11403ee3d2" /> |

susnux commented on apps/files/src/actions/moveOrCopyAction.ts:
```diff
@@ -22,6 +22,8 @@ import logger from '../logger.ts'
 import { getContents } from '../services/Files.ts'
 import { canCopy, canMove, getQueue, MoveCopyAction } from './moveOrCopyActionUtils.ts'
 
+import '@nextcloud/dialogs/style.css'
```
Already be included in server styles - so no need to import

artonge replied:
Doesn't it make sense to keep the import close to the usage?
Or will it import it twice?

susnux replied:
It duplicates as the styles are globally registered already (e.g. we use them OC and OCP API).

In apps they also only need to be imported once per entry point, for server this is globally done in the server styles.

artonge committed:
fix(FilesPicker): Prevent selection of nodes without create permission

And remove the disable callback for the button as it is not needed anymore.

This gives a better experience as the nodes do not have a pointer for cursor, and the submit button does not react when clicking on them.

Signed-off-by: Louis Chmn <louis@chmn.me>

artonge commented:
> Then you should be able to navigate into those greyed folder but not be able to pick them.

No, for E2EE it does not make sense as we cannot even load their content. But luckily, they are also not readable, I'll push a better approach in a few minutes.

artonge committed:
fix(FilePicker): Modernise dav imports

Signed-off-by: Louis Chmn <louis@chmn.me>

jancborchardt reviewed (approved):
Looks good!

Might additionally be good to have some textual support there, like a title on hover & focus which says "No permissions to upload to this folder", for example.

artonge committed:
chore(deps): Bump @nextcloud/dialogs to v7.1.0

Signed-off-by: Louis Chmn <louis@chmn.me>

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

nextcloud-command committed:
chore(assets): Recompile assets

Signed-off-by: nextcloud-command <nextcloud-command@users.noreply.github.com>

artonge commented:
/backport cb062d26915ddc76ca246e2e88f7d785466e82b1 to stable31
